src/llm/llm_executor.hpp

```diff
@@ -0,0 +1,47 @@
+//*****************************************************************************
+// Copyright 2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <atomic>
+#include <memory>
+#include <thread>
+
+#include <continuous_batching_pipeline.hpp>
+
+class LLMExecutor {
+    std::atomic<bool> end;
+    std::shared_ptr<ContinuousBatchingPipeline> pipe;
+    std::thread llmExecutorThread;
+
+    static void run(std::shared_ptr<ContinuousBatchingPipeline> pipe, std::atomic<bool>* end) {
+        while (!(*end)) {
+            pipe->step();
```

Comment:
Is this step blocking if there are no iterations to make? (0 requests processing)
If so, we may have a deadlock on line 45 (waiting for the thread to join will be forever)

Reply:
It's not blocking. But perhaps we could add early skip inside step() to avoid running anything when we don't have active or awaiting requests.

Reply:
@mzegla with some sleep in between skips? otherwise we are burning cpu cycles

Reply:
With simple sleep we would risk underutilization. Some notification solution would be better.